### PR TITLE
ESP-221_BasicAuth - adding HeaderPreprocessor and support for Basic Authentication mechanism.

### PR DESCRIPTION
**Pull request by kristoffSC**
Signed-off-by: Krzysztof Chmielewski <krzysztof.chmielewski@getindata.com>

#### Description
Add Http Header value preprocessor mechanism, that can preprocess defined header value before setting it on the request.
It allows user to specify `Autorization` header for Basic Authentication. The value will be converted to Base64,
  or it will be used is if the valued is starting from `Basic ` prefix.

##### PR Checklist
- [x] Tests added
- [x] [Changelog](CHANGELOG.md) updated

### CODE REVIEW

File: src/main/java/com/getindata/connectors/http/internal/utils/ConfigUtils.java [hunk 1]
```diff
@@ -81,34 +80,6 @@ public static String extractPropertyLastElement(String propertyKey) {
...
-        return headerMap
-            .entrySet()
-            .stream()
-            .flatMap(entry -> {
-                String originalKey = entry.getKey();
-                String newKey = ConfigUtils.extractPropertyLastElement(originalKey);
-                return Stream.of(newKey, entry.getValue());
-            }).toArray(String[]::new);
-    }
-
```
Comment (kristoffSC) [inline src/main/java/com/getindata/connectors/http/internal/utils/ConfigUtils.java:111]: Moved to HttpHeaderUtils.java

File: src/test/java/com/getindata/connectors/http/internal/utils/ConfigUtilsTest.java [hunk 2]
```diff
@@ -94,11 +94,11 @@ public void shouldThrowOnInvalidProperty(String invalidProperty) {
     public void flatMapPropertyMap() {
         Map<String, String> propertyMap = Map.of(
             "propertyOne", "val1",
-            "my.propertyTwo", "val2",
-            "my.super.propertyThree", "val3"
+            "propertyTwo", "val2",
+            "propertyThree", "val3"
         );
```
Comment (kristoffSC) [inline src/test/java/com/getindata/connectors/http/internal/utils/ConfigUtilsTest.java:99]: after changes in HttpHeaderUtils, the `toHeaderAndValueArray` is no longer responsible to picking up the header name from property name.

It's job now is only convert map to array

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,12 +4,15 @@
 
 ### Added
 
+- Add Http Header value preprocessor mechanism, that can preprocess defined header value before setting it on the request.
+- Allow user to specify `Autorization` header for Basic Authentication. The value will be converted to Base64,
+  or it will be used is if the valued is starting from `Basic ` prefix.
```
Comment (grzegorz8) [inline CHANGELOG.md:9]: starting *with* ? The same in README.md:237

File: src/main/java/com/getindata/connectors/http/internal/HeaderValuePreprocessor.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package com.getindata.connectors.http.internal;
+
+import java.io.Serializable;
+
+/**
+ * Processor interface that its job is to modify header value based on implemented logic.
```
Comment (grzegorz8) [inline src/main/java/com/getindata/connectors/http/internal/HeaderValuePreprocessor.java:6]: Maybe `Processor interface which modifies ...` ?

File: CHANGELOG.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (kristoffSC) [reply CHANGELOG.md]: fixed

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,12 +4,15 @@
 
 ### Added
 
+- Add Http Header value preprocessor mechanism, that can preprocess defined header value before setting it on the request.
+- Allow user to specify `Autorization` header for Basic Authentication. The value will be converted to Base64,
```
[the remark this answers is not in the document]
Comment (kristoffSC) [reply CHANGELOG.md]: fixed

**Review by swtwsk — APPROVED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (kristoffSC) [reply CHANGELOG.md]: Fixed

**Pull request merged**